sycl/source/detail/error_handling/error_handling.cpp

```diff
@@ -360,5 +360,27 @@ void handleErrorOrWarning(pi_result Error, const device_impl &DeviceImpl,
 }
 
 } // namespace detail::enqueue_kernel_launch
+
+namespace detail::kernel_get_group_info {
+void handleErrorOrWarning(pi_result Error, pi_kernel_group_info Descriptor) {
+  assert(Error != PI_SUCCESS &&
+         "Success is expected to be handled on caller side");
+  switch (Error) {
+  case PI_ERROR_INVALID_VALUE:
+    if (Descriptor == CL_KERNEL_GLOBAL_WORK_SIZE)
+      throw sycl::exception(
+          sycl::make_error_code(errc::invalid),
+          "info::kernel_device_specific::global_work_size descriptor may only "
+          "be used if the device type is device_type::custom or if the kernel "
+          "is a built-in kernel.");
+    break;
+  // TODO: Handle other error codes
+  default:
+    throw runtime_error(
+        "Native API failed. Native API returns: " + codeToString(Error), Error);
```

Comment:
Preferably we would use `checkPiResult` here. Not only to have fewer maintenance points, but also to leverage the handling of `PI_ERROR_PLUGIN_SPECIFIC_ERROR`.

```suggestion
    Plugin.checkPiResult(Error);
    break;
```

Note that `checkPiResult` may elect not to throw in case the error code is actually just a warning.